cpp/dolfinx/io/XDMFFile.cpp

```diff
@@ -334,8 +334,9 @@ template void XDMFFile::write_meshtags(const mesh::MeshTags<std::int32_t>&,
 /// @endcond
 //-----------------------------------------------------------------------------
 mesh::MeshTags<std::int32_t>
-XDMFFile::read_meshtags(const mesh::Mesh<double>& mesh, std::string name,
-                        std::string xpath)
+XDMFFile::read_meshtags_by_name(const mesh::Mesh<double>& mesh,
```

Comment:
Do we really want a separate function for this in C++? Couldn't we just put the attribute name as a fourth argument (optional) argument.

I don't see the added value of a constructor for handling one optional value to have a backwards compatible C++ interface. @garth-wells

Reply:
Agree, one function would be better. Would is make sense to use `std::optional` for `attribute_name`?

Related to this, we should not pass empty strings as 'null' - use `std::optional`.